src/App.jsx

```diff
@@ -14,33 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ABC = 'Sort alphabetically';
+const SORT_FIELD_LENGTH = 'Sort by length';
+const SORT_FIELD_RESET = 'Reset';
+const SORT_FIELD_REVERSE = 'Reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [currentSort, setCurrentSort] = useState('');
+  const [reversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  let sortedGoods = [...goodsFromServer];
+  const isReset = currentSort || reversed;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const handleReset = () => {
+    setCurrentSort('');
+    setReversed(false);
+  };
+
+  switch (currentSort) {
+    case SORT_FIELD_ABC:
+      sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SORT_FIELD_LENGTH:
+      sortedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    case SORT_FIELD_RESET:
+      sortedGoods = [...goodsFromServer];
+      break;
+    default:
+      break;
+  }
+
+  if (reversed) {
+    sortedGoods = sortedGoods.toReversed();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', {
+            'is-light': currentSort !== SORT_FIELD_ABC,
+            'is-info': currentSort === SORT_FIELD_ABC,
+          })}
+          onClick={() => setCurrentSort(SORT_FIELD_ABC)}
+        >
+          {SORT_FIELD_ABC}
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button', {
+            'is-light': currentSort !== SORT_FIELD_LENGTH,
+            'is-success': currentSort === SORT_FIELD_LENGTH,
+          })}
```

Comment:
There is something wrong with tbutton'sons colors, they are all grey in your case
![image](https://github.com/user-attachments/assets/114ee825-cbcf-42fe-80c3-71ca7408f70f)

Reply:
> There is something wrong with tbutton'sons colors, they are all grey in your case ![image](https://private-user-images.githubusercontent.com/63243136/369708404-114ee825-cbcf-42fe-80c3-71ca7408f70f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNzI4OTIsIm5iZiI6MTcyNzA3MjU5MiwicGF0aCI6Ii82MzI0MzEzNi8zNjk3MDg0MDQtMTE0ZWU4MjUtY2JjZi00MmZlLTgwYzMtNzFjYTc0MDhmNzBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDA2MjMxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwMjhjYjE3YmNiNDAyN2JiZGQ0Y2FiMDBiOTM1Y2JlNWMyN2E5NDg4MTY0M2ZkMjRmYTFlMjNkZjE1YWVhYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wVZrIVvQLbW-9vlXCLWtN-TM59HkWhXUkoRuWmlMvfU)

You still have a problem with buttons colors